### PR TITLE
Add NNCF quantization for CLIP notebook

### PR DESCRIPTION
**Pull request by l-bat**
Add post-training optimization support by NNCF for CLIP model from [228-clip-zero-shot-image-classification.ipynb](https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb)

### CODE REVIEW

**Comment by l-bat** [comment]
@KodiaqQ I have to disable SmoothQuant algorithm for the CLIP model due to significant accuracy drop.
Please confirm that the SmoothQuant algorithm works correctly.

**Comment by l-bat** [comment]
@eaidova do you mind if I move `visualize_result` function to a python file to avoid copy paste in quantization notebook?

**Comment by eaidova** [comment]
> @eaidova do you mind if I move `visualize_result` function to a python file to avoid copy paste in quantization notebook?

I think, no problem, sure

**Comment by l-bat** [comment]
@MaximProshin please take a look

File: notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb:11]: I would slightly rephrase it and say like "The goal of this tutorial is to demonstrate how to speed up the model by applying 8-bit post-training quantization from [NNCF](https://github.com/openvinotoolkit/nncf/) (Neural Network Compression Framework)", i.e. switch the focus from NNCF to the optimization

File: notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb:18]: models that are used => model that is used

File: notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb:81]: Why is this cell needed if you use the 1st notebook results as input?

File: notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb:179]: I get an error in this step (suppose it's due to Intel politics and the dataset is not loaded):

RuntimeError: DataLoader worker (pid(s) 11696, 16624) exited unexpectedly

Comment (l-bat) [reply notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb:179]: Removed `num_workers=2` from `torch.utils.data.DataLoader`

**Comment by MaximProshin** [comment]
LGTM, with recent changes in the notebook and NNCF from develop, it works fine. @l-bat , please specify nncf version in the notebook and remove the ignored scope as we discussed offline. After that I'm ok to merge it.
@eaidova , please also take a look or suggest who can do it from the notebooks side. 
cc @alexsu52 @AlexKoff88 FYI

**Review by MaximProshin — APPROVED** [review]

File: notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-quantize-nncf.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    fp16_model_path = 'clip-vit-base-patch16.xml'
</pre><p>should we chack that this file really exists?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

**Comment by eaidova** [comment]
@l-bat can we have also some sorting of notebooks on naming level? Now, the quantization notebook in directory structure is first, there is some risk from the user experience side that the user without reading readme starts to try it from the beginning of the folder, without running first one. maybe we can rename them to 228-clip-zero-shot-convert and 228-clip-zero-shot-quantize? (or any other names just keep common name and suffixes convert and quantize for both parts)

**Comment by eaidova** [comment]
@l-bat  please fix link in readme after notebook renaming:
/home/runner/work/openvino_notebooks/openvino_notebooks/notebooks/228-clip-zero-shot-image-classification/README.md: invalid URL reference '228-clip-quantize.ipynb'

your TODO is still actual and should be applied in this notebook or it is related to image bind?
Can we merge this after link fix?

**Comment by l-bat** [comment]
> your TODO is still actual and should be applied in this notebook or it is related to image bind?

No, remove it

> Can we merge this after link fix?

Yes, link has been fixed

File: notebooks/228-clip-zero-shot-image-classification/README.md
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/README.md:18]:
```suggestion
The first notebook contains the following steps:
```

File: notebooks/228-clip-zero-shot-image-classification/README.md
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/README.md:24]:
```suggestion
The second notebook contains the following steps:
```

File: notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-convert.ipynb
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-convert.ipynb:263]:
```suggestion
    "Open the [228-clip-zero-shot-quantize](228-clip-zero-shot-quantize.ipynb) notebook to quantize the IR model with the Post-training Quantization API of NNCF and compare `FP16` and `INT8` models."
```

File: notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-quantize.ipynb
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-quantize.ipynb:18]:
```suggestion
    "> **NOTE**: you should run [228-clip-zero-shot-convert](228-clip-zero-shot-convert.ipynb) notebook first to generate OpenVINO IR model that is used for quantization."
```

File: notebooks/228-clip-zero-shot-image-classification/README.md
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/README.md:26]:
```suggestion
2. Check the model result using the same input data from the [first notebook](228-clip-zero-shot-convert.ipynb).
```

File: notebooks/228-clip-zero-shot-image-classification/README.md
Comment (eaidova) [inline notebooks/228-clip-zero-shot-image-classification/README.md:30]:
```suggestion
NNCF performs quantization within the OpenVINO IR. It is required to run the first notebook before running the second notebook.
```

**Pull request merged**
